### PR TITLE
Feature/improved functionality android

### PR DESCRIPTION
**Pull request by aca-mobile**
Hi,

Recently, we needed to support BLE in one of our apps. Because of missing features and different implementation between iOS and Android, we decided to change the Android implementation to keep it consistent with the API which was already well defined for iOS. 

It's not complete yet because we currently don't have the needs to let the device act as a peripheral, so for now we didn't implement these features. Feel free to extend Android functionality. This PR does not contain all the features that are already available for iOS, but it is still an improvement compared to the Android-implementation that is available right now.

Documentation is now available in the /documentation folder and can be edited by contributors when they change the implementation. It might be a good idea to remove the wiki and refer to the documentation folder instead.

### CODE REVIEW

File: android/manifest [hunk 1]
```diff
@@ -15,4 +15,4 @@ name: ti.bluetooth
 moduleid: ti.bluetooth
 guid: 1d5352ed-3084-4f88-9f02-8e7be75f4ca9
 platform: android
-minsdk: 6.0.0.GA
+minsdk: 7.5.0
```
Comment (m1ga) [inline android/manifest]: make this 7.0.0? Shouldn't break anything and would make it more useful for people with existing apps.

**Comment by m1ga** [comment]
Awesome job @aca-mobile :+1: ! Could you run `clang-format -style=file -i [JAVA FILE]` so they are formatted the same. And perhaps lower the minimum SDK version to 7.0.0.
I try to test it at the weekend!

**Comment by hansemannn** [comment]
Thanks for the update! Does the current example now work 1:1 on Android? Or should a platform-specific example be added? This is great effort, thanks again @aca-mobile !

**Comment by aca-mobile** [comment]
There might be some events that won't be triggered on Android, and there's no PeripheralManager available, but the basic discovering devices/connecting device/connection error/find services/characteristics/write on characteristic and get notified when something has been written are working fine for us. Not a single OS_ANDROID or OS_IOS is needed in our flow :)

If needed, I'll provide an Android-specific example, but I think that the documentation is clear enough :)

**Comment by hansemannn** [comment]
Nice! I think we are good here. Merging once @m1ga approved as well.

**Comment by m1ga** [comment]
I've tested it on Android.
I think it would be good to add at least `Ti.API.info(e.peripheral.name);` and a connect to the `didDiscoverPeripheral` event since the output will be just `Object` and not the actual paramaters that are returned. 

Also add `if (OS_IOS) {...}` around the `createPeripheralManager` and the events at the bottom (makes it quicker to use the example).

The options in `connectPeripheral` are not optional on Android.

File: example/app.js [hunk 2]
```diff
@@ -1 +1,202 @@
...
+    backgroundColor: '#fff'
+});
+
+var btn1 = Ti.UI.createButton({
+    title: 'Start scan',
+    top: 40
+});
+
+var centralManager = BLE.createCentralManager();
+var peripheralManager = BLE.createPeripheralManager();
```
Comment (m1ga) [inline example/app.js]: `if (OS_IOS) {...}`

File: example/app.js [hunk 3]
```diff
@@ -1 +1,202 @@
...
+    Ti.API.info(e);
+});
+
+centralManager.addEventListener('didFailToConnectPeripheral', function(e) {
+    Ti.API.info('didFailToConnectPeripheral');
+    Ti.API.info(e);
+});
+
+/** 
+ * Peripheral Manager Events
```
Comment (m1ga) [inline example/app.js]: put them all into a `if (OS_IOS) {...}`

File: android/src/ti/bluetooth/central/TiBluetoothCentralManagerProxy.java [hunk 4]
```diff
@@ -0,0 +1,232 @@
...
+
+  @Kroll.method
+  public void stopScan() {
+    bluetoothAdapter.getBluetoothLeScanner().stopScan(scanCallback);
+    isScanning = false;
+  }
+
+  @Kroll.method
+  public void connectPeripheral(TiBluetoothPeripheralProxy peripheral,
+                                KrollDict properties) {
```
Comment (m1ga) [inline android/src/ti/bluetooth/central/TiBluetoothCentralManagerProxy.java]: the properties should be `(optional = true)`

File: example/app.js [hunk 5]
```diff
@@ -1 +1,202 @@
...
+    centralManager.stopScan();
+});
+
+/** 
+ * Central Manager Events
+ */
+
+centralManager.addEventListener('didDiscoverPeripheral', function(e) {
+    Ti.API.info('didDiscoverPeripheral');
+    Ti.API.info(e);
```
Comment (m1ga) [inline example/app.js]: perhaps add
```
Ti.API.info('Connect to ' + e.peripheral.name);
	centralManager.connectPeripheral(e.peripheral, {
		notifyOnConnection: true,
		notifyOnDisconnection: true
	});
```
so people see some real output

File: example/app.js (hunk 2 above)
Comment (aca-mobile) [reply example/app.js]: Done!

File: example/app.js (hunk 3 above)
Comment (aca-mobile) [reply example/app.js]: Done!

File: android/src/ti/bluetooth/central/TiBluetoothCentralManagerProxy.java (hunk 4 above)
Comment (aca-mobile) [reply android/src/ti/bluetooth/central/TiBluetoothCentralManagerProxy.java]: Done!

File: example/app.js (hunk 5 above)
Comment (aca-mobile) [reply example/app.js]: Done!

**Comment by m1ga** [comment]
Awesome! LGTM.
Thanks @aca-mobile

**Pull request merged**